panic at `pgwatch3/log/log_broker_hook.go`

**Describe the bug**
pgwatch3 die after a few minutes of idle
 

**Steps to produce the behavior:**
1. Follow the readme to add test db and bench
2. poke around the webui and wait for a while?
3. panic

~~I haven't been able to reproduce it though~~ reproduced (but the situation is kinda like iterator's use after invalidation)

**Screenshots**
~~I wonder if its also related to the sql error above~~
Not related but weird
<img width="956" alt="image" src="https://github.com/cybertec-postgresql/pgwatch3/assets/29861494/b9af3d9f-9abb-4782-898b-7331ac21bd52">

```
panic: runtime error: index out of range [1] with length 1

goroutine 902 [running]:
github.com/cybertec-postgresql/pgwatch3/log.(*BrokerHook).RemoveSubscriber(0xc0000a7110, 0xc000618420)
	/pgwatch3/log/log_broker_hook.go:59 +0x205
github.com/cybertec-postgresql/pgwatch3/webserver.writer(0xc0000b3080, {0x1638860, 0xc00003ebd0})
	/pgwatch3/webserver/wslog.go:64 +0x2a2
created by github.com/cybertec-postgresql/pgwatch3/webserver.(*WebUIServer).serveWsLog in goroutine 915
	/pgwatch3/webserver/wslog.go:82 +0x1ba
```

Fix: iterate from back to front so you don't have to worry about indexes that are deleted

--- a/src/log/log_broker_hook.go
+++ b/src/log/log_broker_hook.go
@@ -2,6 +2,7 @@ package log
 
 import (
 	"context"
+	"slices"
 	"sync"
 	"time"
 
@@ -55,11 +56,9 @@ func (hook *BrokerHook) AddSubscriber(msgCh MessageChanType) {
 func (hook *BrokerHook) RemoveSubscriber(msgCh MessageChanType) {
 	hook.mu.Lock()
 	defer hook.mu.Unlock()
-	for i := len(hook.subscribers) - 1; i >= 0; i-- {
-		if hook.subscribers[i] == msgCh {
-			hook.subscribers = append(hook.subscribers[:i], hook.subscribers[i+1:]...)
-		}
-	}
+	hook.subscribers = slices.DeleteFunc(hook.subscribers, func(E MessageChanType) bool {
+		return E == msgCh
+	})
 }
 
 var defaultFormatter = &logrus.TextFormatter{DisableColors: true}

--- a/src/log/log_broker_hook.go
+++ b/src/log/log_broker_hook.go
@@ -55,7 +55,7 @@ func (hook *BrokerHook) AddSubscriber(msgCh MessageChanType) {
 func (hook *BrokerHook) RemoveSubscriber(msgCh MessageChanType) {
 	hook.mu.Lock()
 	defer hook.mu.Unlock()
-	for i := range hook.subscribers {
+	for i := len(hook.subscribers) - 1; i >= 0; i-- {
 		if hook.subscribers[i] == msgCh {
 			hook.subscribers = append(hook.subscribers[:i], hook.subscribers[i+1:]...)
 		}

--- a/src/log/log_broker_hook_test.go
+++ b/src/log/log_broker_hook_test.go
@@ -1,0 +1,25 @@
+package log
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRemoveSubscriber(t *testing.T) {
+	hook := NewBrokerHook(context.Background(), "info")
+	msgChan1 := make(MessageChanType)
+	msgChan2 := make(MessageChanType)
+	hook.AddSubscriber(msgChan1)
+	hook.AddSubscriber(msgChan2)
+	assert.Equal(t, 2, len(hook.subscribers))
+
+	// Remove the first subscriber
+	hook.RemoveSubscriber(msgChan1)
+	assert.Equal(t, 1, len(hook.subscribers))
+
+	// Remove the last subscriber, this is where the "index out of range" error occurs
+	hook.RemoveSubscriber(msgChan2)
+	assert.Equal(t, 0, len(hook.subscribers))
+}

